src/dev/ADCMux.cpp

```diff
@@ -0,0 +1,32 @@
+#include <EVT/io/ADC.hpp>
+#include <EVT/dev/ADCMux.hpp>
+#include <EVT/io/pin.hpp>
+
+namespace EVT::core::DEV {
+
+    ADCMux::ADCMux(EVT::core::IO::GPIO& ctrlGPIO1, EVT::core::IO::GPIO& ctrlGPIO2, IO::ADC& adc) : ctrlGPIO1(ctrlGPIO1), ctrlGPIO2(ctrlGPIO2), adc(adc) {
+    }
+
+    uint32_t ADCMux::getInput() {
+        return adc.readRaw();
+    }
+
+    //Set the output pins based on an input number
+    void ADCMux::setOutputPin(uint8_t number) {
+        if((number & 0x1) == 0 && (number & 0x2) == 0) {
```

Comment:
Id recommend a switch statement here. So it would be
```c
switch(number) {
    case 0:
        ctrlGPIO1.writePin(low);
        ctrlGPIO1.writePin(low);
     case 1:
        ...
}
```

Reply:
just changed thank you